packages/fetch-and-compile/test/fetch.test.ts

```diff
@@ -0,0 +1,130 @@
+import assert from "assert";
+import { describe, it } from "mocha";
+import Config from "@truffle/config";
+import { fetchAndCompile } from "../lib/index";
+import axios from "axios";
+import sinon from "sinon";
+const fixture: any = require("./fixture.js");
+
+describe("fetches contract on mainnet and checks for verification", () => {
+  const config = Config.default().merge({
+    networks: {
+      mainnet: {
+        url: 'https://mainnet.infura.io/v3/',
+        network_id: 1
+      }
+    },
+    network: "mainnet",
+  });
+  //stubbing axios.get method
+  beforeEach(() => {
+    //asserting that mainnet url and contract address is passed as args
+    sinon.stub(axios, 'get').withArgs("https://api.etherscan.io/api", {
+      params: {
+        module: "contract",
+        action: "getsourcecode",
+        address: '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D',
+        apikey: ''
+      },
+      responseType: "json",
+      maxRedirects: 50
+    }).returns(Promise.resolve({ data: fixture.mainnetData }))
+    //@ts-ignore
+    axios.get.callThrough();
+  })
+  afterEach(() => {
+    //@ts-ignore
+    //restoring stub
+    axios.get.restore()
+  })
+  it('resolves with verified contract', () => {
+    return fetchAndCompile('0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D', config).then(result => {
+      let contractName = result.sourceInfo.contractName
+      //assert contract name is correct 
+      assert.equal(contractName, "UniswapV2Router02")
+      //assert contract name is not undefined
+      assert.notEqual(contractName, undefined)
+    })
+  })
+});
+
+describe("fetches contract on arbitrum and checks for verification", () => {
```

Comment:
I think this test module could be cleaned up if all the tests resided under one describe. I don't feel strongly about it, but there is an opportunity to refactor.

Reply:
Definitely I made this change when I refactored above. Thanks!